### ISSUE TITLE
Odd diamond operator for generic interface

### ISSUE THREAD

**Issue by almozavr** [immutables/immutables#362]
## Scope
```java
@Value.Immutable
public interface Entity<T> {
    T item();
}
```
## Problem
Generated class has diamond operator `<>` which blocks 1.6 compatibility
```java
public final class ImmutableEntity<T> implements Entity<T> {
  public final ImmutableEntity<T> withType(Entity.Type type) {
    if (this.type == type) return this;
    Entity.Type newValue = Objects.requireNonNull(type, "type");
    return new ImmutableEntity<>(newValue, this.item); // <----- should be ImmutableEntity<T>
  }

  public final ImmutableEntity<T> withItem(T item) {
    if (this.item == item) return this;
    T newValue = Objects.requireNonNull(item, "item");
    return new ImmutableEntity<>(this.type, newValue); // <----- should be ImmutableEntity<T>
  }  
}
```

**Comment by elucash** [comment]
think we should support java 6 for 2.x line, thank you for pointing out!

**Comment by elucash** [comment]
released in 2.2.7 thank you!
